### DISCUSSION TITLE
Add custom metadata to ChatMessageContent  like DateTimeCreated message was generated

### DISCUSSION THREAD

**Discussion by giannik** [microsoft/semantic-kernel#9996] [Q&A] [answered]
What is the recommended way to add custom information to either the user or ai generated response like datetime created to show the actual time the message was logged in the chat history. This could also apply to any other custom metadata you may want to save

```
_chatHistory.Add(new ChatMessageContent() { Content = input, 
                Role = ChatMessageRole.User
                DateTimestamp = DateTime.UtcNow ??
                
            });;
```

**Comment by sophialagerkranspandey** [comment]
Tagging @markwallace-microsoft to reply once he returns from vacation.

**Comment by markwallace-microsoft** [comment, marked as answer]
@giannik You can use the metadata dictionary to store additional data

```csharp
_chatHistory.Add(new ChatMessageContent() { Content = input, 
                Role = ChatMessageRole.User
                Metadata = <IReadOnlyDictionary containing your metadata>
            });;
```
